app/src/main/java/com/jmstudios/redmoon/service/AccessibilityFilterService.kt

```diff
@@ -0,0 +1,39 @@
+package com.jmstudios.redmoon.service
+
+import android.accessibilityservice.AccessibilityService
+import android.animation.ValueAnimator
+import android.graphics.PixelFormat
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import android.view.accessibility.AccessibilityEvent
+import com.jmstudios.redmoon.*
+import com.jmstudios.redmoon.helper.Logger
+import org.greenrobot.eventbus.Subscribe
+
+
+class AccessibilityFilterService : AccessibilityService() {
+    lateinit var mFilter: Overlay
+
+    override fun onAccessibilityEvent(event: AccessibilityEvent) {}
+    override fun onInterrupt() {}
+
+    override fun onServiceConnected() {
+        mFilter = Overlay(applicationContext)
+        (getSystemService(WINDOW_SERVICE) as WindowManager).addView(mFilter, mFilter.mLayoutParams)
+        mFilter.visibility = View.GONE;
+        EventBus.register(this)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        (getSystemService(WINDOW_SERVICE) as WindowManager).removeView(mFilter)
+        EventBus.unregister(this);
```

Comment:
```suggestion
        EventBus.unregister(this);
        super.onDestroy()
        (getSystemService(WINDOW_SERVICE) as WindowManager).removeView(mFilter)
```
I think you want to avoid receiving any new events while shutting down, although I'm not 100% sure it will cause issues.

Reply:
JavaScript has rotted my brain on concurrency, good catch.